web/packages/teleport/src/Sessions/SessionList/SessionJoinBtn.tsx

```diff
@@ -35,8 +35,14 @@ export const SessionJoinBtn = ({
   participantModes: ParticipantMode[];
   showCTA: boolean;
 }) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement>(null);
```

Comment:
A feedback for the future as this pattern wasn't introduced by this PR specifically, but I don't think it's a good practice to store HTML elements in the state – you might end up storing an element in the state that's no longer in DOM. Was this done as a workaround for some other problem?

Looking at the previous version of `JoinMenu`, I think a more idiomatic solution would be to create a new ref with `useRef`, pass it as the `ref` prop to `<ButtonBorder>`. Then to control whether `<Menu>` is supposed to be open or not, we could have `const [isOpen, setIsOpen] = useState(false)` and on the button click we could flip it to true.

Reply:
[`MenuLogin`](https://github.com/gravitational/teleport/blob/2bb38605136375803416ebf10c7437e80b5938d4/web/packages/shared/components/MenuLogin/MenuLogin.tsx#L80) does it this way already. It looks like the prop has to be named `setRef` for the button components, I'm not exactly sure why it's done like this.

https://github.com/gravitational/teleport/blob/2bb38605136375803416ebf10c7437e80b5938d4/web/packages/design/src/Button/Button.jsx#L23-L25